keystrokes/src/components/Character/CharacterDetails/ManualKeybindModal.js

```diff
@@ -0,0 +1,251 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Button, Grid, TextField, MenuItem } from '@material-ui/core';
+import update from 'immutability-helper';
+import PropTypes from 'prop-types';
+import KeybindTable from './KeybindTable';
+import styleGuide from '../../../stylesheets/style';
+import { verifyKey, validatePress } from '../../utils/utils';
+import {
+	ref,
+	targetting,
+	mods,
+	characterDetails,
+} from '../../../config/constants';
+
+ManualKeybindModal.propTypes = {
+	isOpen: PropTypes.bool.isRequired,
+	setIsOpen: PropTypes.func.isRequired,
+	characterClass: PropTypes.string.isRequired,
+	characterSpec: PropTypes.number.isRequired,
+	setAllKeybindings: PropTypes.func.isRequired,
+	allKeybindings: PropTypes.object.isRequired,
+	keyBindingKey: PropTypes.string.isRequired,
+};
+
+function ManualKeybindModal({
+	isOpen,
+	setIsOpen,
+	characterClass,
+	characterSpec,
+	setAllKeybindings,
+	allKeybindings,
+	keyBindingKey,
+}) {
+	const classes = styleGuide();
+	const [keybinding, setKeybinding] = useState({
+		Spell: null,
+		Target: null,
+		Mod: null,
+		Key: null,
+	});
+	const [allKeybinds, setAllKeybinds] = useState([]);
+	const [loading, setLoading] = useState(true);
+	const [Spells, setSpells] = useState();
+	const spec = characterDetails.class[characterClass][
+		characterSpec
+	].toUpperCase();
+
+	useEffect(() => {
+		async function getSpells() {
+			const path = `/Spells/${characterClass}`;
+			const snapShot = await ref.child(path).once('value');
+			if (snapShot.exists()) {
+				setSpells(snapShot.val());
+				setLoading(false);
+			}
+		}
+		getSpells();
+	}, []);
```

Comment:
from a quick glance it looks like you're missing the characterClass dependancy on the useEffect. There's a great plugin to show linting warnings for useEffect